### DISCUSSION TITLE
Can cy.wrap() be retried?

### DISCUSSION THREAD

**Discussion by romankhomitskyi** [cypress-io/cypress#26462] [Questions and Help]
```
Cypress.Commands.addQuery('getInnerText', () => {
	return (subject): Chainable<string> => subject.get(0).innerText;
});

Cypress.Commands.add(
	'assertInnerText',
	{
		prevSubject: true,
	},
	(subject, text: string) => {
		cy.wrap(subject).getInnerText().should('equal', text);
	}
);
```

I have the following code, but cy.wrap(subject) is not retried

**Comment by portyu9** [comment]
Hi @romankhomitskyi 

`cy.wrap()` can participate in retryable assertions, but it does **not** make an already-captured DOM subject get queried from the application again.

That is the issue with this pattern: `assertInnerText` is a normal custom command, so by the time it receives `subject`, the original element has already been captured. Wrapping that subject again cannot rerun the original `cy.get()`.

Your custom query can instead stay directly linked to the original query:

```js
cy.get('[data-cy=item]')
  .getInnerText()
  .should('equal', 'expected text')
```

Because `getInnerText` is a query, Cypress can retry the linked query chain when the assertion fails.

For a reusable assertion, another option is to make the whole operation a custom query:

```js
Cypress.Commands.addQuery('assertInnerText', function (text) {
  return (subject) => {
    Cypress.ensure.isElement(subject, 'assertInnerText', cy)

    expect(subject.get(0).innerText).to.eq(text)

    return subject
  }
})
```

Then:

```js
cy.get('[data-cy=item]').assertInnerText('expected text')
```

One other detail: the inner function of your `getInnerText` query returns a `string`; it does not return `Chainable<string>`. Cypress also recommends a normal `function` for the outer `addQuery()` callback because the query API uses its callback context.

The key is to keep the original DOM query, derived query, and assertion in the same retryable query chain rather than introducing a normal custom command around an already-captured subject.

Please mark this answer as accepted if it helped, thank you.
